source/yats/include/yats/identifier.h

```diff
@@ -1,71 +1,174 @@
     {
-        return character - 'a';
+        if (length == 0)
+            return 0;
+
+        // The length of a string is limited to 12 characters excluding the termination character.
+        constexpr auto max_length = 12;
+        const auto truncated_length = std::min(static_cast<int>(length), max_length);
+
+        auto value = char_to_int(string[truncated_length - 1]);
+
+        // Highest 60 bits encode the string with up to 12 characters from an 32 character alphabet.
+        // Highest 5 bits contain last character in string.
+        for (auto i = truncated_length - 2; i >= 0; --i)
+        {
+            value = (value << width_of_character) | char_to_int(string[i]);
+        }
+
+        // Lowest 4 bits encode the length of the string.
+        value = (value << width_of_length) | truncated_length;
+
+        return value;
     }
 
-    // Special characters
-    constexpr int unique_begin = 26;
-    switch (character)
+    /**
+     * Converts {@code id} to a string. This function is the inverse to {@link identifier#string_to_id(const char*, size_t) string_to_id}.
+     * <p>
+     * Note: Keep in mind that the hashing process is not lossless.
+     * </p>
+     * @param id Id to convert to a string.
+     * @return String
+     */
+    static std::string id_to_string(uint64_t id)
     {
-    case ' ':
-        return unique_begin + 0;
-    case '*':
-        return unique_begin + 1;
-    case '/':
-        return unique_begin + 2;
-    case '&':
-        return unique_begin + 3;
-    case '#':
-        return unique_begin + 4;
-    default:
-    case '_':
-        return unique_begin + 5;
-    }
-}
+        std::string s;
 
-constexpr uint64_t name(const char* string, size_t length)
-{
-    uint64_t value = 0;
+        // lowest 4 bits = length;
+        const int length = id & 0b1111;
+        // cut off length (last 4 bits)
+        id = id >> width_of_length;
 
-    // The length of a string is limited to 12 characters excluding the termination character.
-    constexpr int max_length = 12;
-    const int truncated_length = std::min(static_cast<int>(length), max_length);
+        s.reserve(length);
 
-    // Highest 60 bits encode the string with up to 12 characters from an 32 character alphabet defined by the lookup function.
-    for (int i = max_length - 1; i >= 0; --i)
-    {
-        if (i <= truncated_length)
+        const uint64_t mask = 0b11111;
+
+        for (auto i = 0; i < length; ++i)
         {
-            value = value | lookup(string[i]);
+            const auto char_as_int = id & mask;
+            s.push_back(int_to_char(char_as_int));
+            id = id >> width_of_character;
         }
-        value = value << 5;
+        return s;
     }
 
-    // Lowest 4 bits encode the length of the string.
-    value = value | truncated_length;
+private:
+    /**
+     * Prevents class from being instantiated.
+     */
+    identifier() = default;
```

Comment:
remove this. We do not care if this class can be instantiated by the user. It makes no sense but it does not harm so it is okay.

Reply:
Also in util.h there are lots of structs that can be instantiated but it is never necessary.